File: frontend/src/services/authService.ts
import api from './api';
import { LoginRequest, AuthResponse, UserDto } from '../types';

export const authService = {
  login: async (credentials: LoginRequest): Promise<AuthResponse> => {
    const response = await api.post<AuthResponse>('/auth/login', credentials);
    return response.data;
  },
  register: async (user: UserDto): Promise<UserDto> => {
    const response = await api.post<UserDto>('/auth/register', user);
    return response.data;
  },
};

File: backend/src/main/java/com/healthcare/management/service/AuthService.java
package com.healthcare.management.service;

import com.healthcare.management.dto.AuthResponse;
import com.healthcare.management.dto.LoginRequest;
import com.healthcare.management.dto.UserDto;

public interface AuthService {
    AuthResponse login(LoginRequest loginRequest);
    UserDto registerUser(UserDto userDto);
}
